docs/source/guide.rst

```diff
@@ -39,14 +39,128 @@ The hook will retrieve a balance of an ERC20 token of the provided address.
     return tokenBalance
   }
 
+Another example is useTokenAllowance hook. Instead of balanceOf we use allowance on ERC20 interface.
+
+.. code-block:: javascript
+
+  function useTokenAllowance(
+    ownerAddress: string | Falsy,
+    spenderAddress: string | Falsy
+    tokenAddress: string | Falsy,
+  ) {
+    const [allowance] =
+      useContractCall(
+        ownerAddress &&
+          spenderAddress &&
+          tokenAddress && {
+            abi: ERC20Interface,
+            address: tokenAddress,
+            method: 'allowance',
+            args: [ownerAddress, spenderAddress],
+          }
+      ) ?? []
+    return allowance
+  }
+
+
 The *useContractCall* hook will take care of updating the balance of new blocks.
 The results are deferred so that the hook does not update too frequently.
 
+In our custom hooks we can use any standard react hooks, custom react hooks and useDApp hooks.
+Rules of hooks apply.
+
 All core hooks are available `here <https://github.com/EthWorks/useDapp/tree/master/packages/core/src/hooks>`_.
 
 
 Testing hooks
 *************
 
-Coming soon...
+Let's take useTokenAllowance as an example.
+
+To write a test, start with a setup code that will create a mock provider and test wallets.
+
+.. code-block:: javascript
+
+  const mockProvider = new MockProvider()
+  const [deployer, spender] = mockProvider.getWallets()
+
+Before each test deploy an ERC20 contract. It's important as otherwise the result of one 
+test could break the other one.
+
+.. code-block:: javascript
+
+  let token: Contract
+
+  beforeEach(async () => {
+    const args = ['MOCKToken', 'MOCK', deployer.address, utils.parseEther("10")]
+    token = await deployContract(deployer, ERC20Mock, args)
+  })
+
+After setup, we have to test the hook.
+
+.. code-block:: javascript
+
+  await token.approve(spender.address, utils.parseEther('1'))
+
+  const { result, waitForCurrent } = await renderWeb3Hook(
+    () => useTokenAllowance(deployer.address, spender.address, token.address),
+    {
+      mockProvider,
+    }
+  )
+  await waitForCurrent((val) => val !== undefined)
+
+  expect(result.error).to.be.undefined
+  expect(result.current).to.eq(utils.parseEther('1'))
+
+To check if the hook reads data correctly we need to prepare it first. We approve the spender so then we can check that our hook returns the correct value.
+
+To test the hook we need to render it using renderWeb3Hook. It works like `renderHook` from react-testing library,
+but it wraps the hook into additional providers.
+
+On blockchain everything is async. To make tests predictable we need to wait for the result. 
```

Comment:
With react components updates happens asynchronically, as well as getting updates form blockchain. To get the first value, wait for value to be set.